### PR TITLE
Add NFS storage class for RocketMQ data persistence

### PR DESCRIPTION
**Pull request by liuruiyiyang**
Change List:
1.  Use Statefulset as the basic resource of both Broker and MetaService CR for data storage.
2. Add storage class framework for RocketMQ broker logs and store and namesrv logs based on NFS.
3. Improve the MetaService auto-scaling control logic. (commented a potential solution for now)
4. Update README.md docs to introduce more information about storage class.
5. Rename apiVersion to ```rocketmq.apache.org/v1alpha1```

### CODE REVIEW

**Comment by liuruiyiyang** [comment]
Sure, I will rename the api version and package name.

File: pkg/controller/broker/broker_controller.go [hunk 1]
```diff
@@ -117,57 +119,79 @@ func (r *ReconcileBroker) Reconcile(request reconcile.Request) (reconcile.Result
...
 			if err != nil {
-				reqLogger.Error(err, "Failed to create new Deployment of "+cons.BrokerClusterPrefix+strconv.Itoa(brokerClusterIndex), "Deployment.Namespace", dep.Namespace, "Deployment.Name", dep.Name)
+				reqLogger.Error(err, "Failed to create new StatefulSet of "+cons.BrokerClusterPrefix+strconv.Itoa(brokerClusterIndex), "StatefulSet.Namespace", dep.Namespace, "StatefulSet.Name", dep.Name)
 			}
 		} else if err != nil {
-			reqLogger.Error(err, "Failed to get broker master Deployment.")
+			reqLogger.Error(err, "Failed to get broker master StatefulSet.")
 		}
+
+		// The following code will restart all brokers to update NAMESRV_ADDR env
```
Comment (vongosling) [inline pkg/controller/broker/broker_controller.go]: If no live update feature introduced, please prioritize the rolling update.

File: README.md [hunk 2]
```diff
@@ -3,29 +3,74 @@
...
+RocketMQ Operator is to manage RocketMQ service instances deployed on the Kubernetes cluster.
 It is built using the [Operator SDK](https://github.com/operator-framework/operator-sdk), which is part of the [Operator Framework](https://github.com/operator-framework/).
 
 ## Quick Start
 
+### Prepare Storage Class
+
+The first step is to prepare a storage class to create PV and PVC where the RocketMQ data will be stored. Here we use NFS as the storage class.
+
+1. Deploy NFS server and clients on your Kubernetes cluster. Please make sure they are functional before you go to the next step.
```
[the remark this answers is not in the document]
Comment (liuruiyiyang) [reply README.md]: I will add some instructions about the NFS verification to the README doc

File: README.md [hunk 3]
```diff
@@ -79,15 +137,15 @@ Check the status:
...
-meta-service-d4f5796d-kr2qg         1/1     Running   0          8s      192.168.130.33   k2data-13   <none>           <none>
-rocketmq-operator-564b5d75d-qnpts   1/1     Running   1          3m33s   10.244.1.68      k2data-13   <none>           <none>
+NAME                                      READY   STATUS    RESTARTS   AGE     IP               NODE        NOMINATED NODE   READINESS GATES
+meta-service-0                            1/1     Running   0          3m18s   192.168.130.33   k2data-13   <none>           <none>
+nfs-client-provisioner-7cf858f754-7vxmm   1/1     Running   0          150m    10.244.2.114     k2data-14   <none>           <none>
+rocketmq-operator-564b5d75d-jllzk         1/1     Running   0          5m53s   10.244.2.116     k2data-14   <none>           <none>
 ```
 
-We can see that there are 2 name service Pods running on 2 nodes and their IP addresses. Modify the ```nameServers``` field in the ```cache_v1alpha1_broker_cr.yaml``` file using the IP addresses.
+We can see that there are 1 name service Pods running on 1 nodes and their IP addresses. Modify the ```nameServers``` field in the ```cache_v1alpha1_broker_cr.yaml``` file using the IP addresses.
```
[the remark this answers is not in the document]
Comment (liuruiyiyang) [reply README.md]: I unified the name to NameService, no more MetaService now

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (vongosling) [reply README.md]: Great~

**Pull request merged**